### DISCUSSION TITLE
How to extracting table content without bottom border

### DISCUSSION THREAD

**Discussion by Taootrum** [jsvine/pdfplumber#631] [Ask for help with specific PDFs]
I have a[ pdf ](https://github.com/jsvine/pdfplumber/files/8342262/Table_Example_ori.pdf)that contains multiple tables, but some tables are spread across pages and have no border at the bottom. When parsing, the row of data without the bottom border will be lost.
My Code:

```
with pdfplumber.open("Table_Example_ori.pdf") as pdf:
    page = pdf.pages[0]
    tables = page.extract_tables()
    print(tables)
```
such as:  Which line of frequency_mode will be lost
![Snipaste_2022-03-24_22-26-24](https://user-images.githubusercontent.com/37130412/159938259-786cd81a-dce1-4398-a029-9e9a1759dad7.png)

Please help to answer this question, thank you very much

**Comment by jsvine** [comment]
Hi @Taootrum, and thanks for your interest in this library. I believe that the response to the following discussion may help: https://github.com/jsvine/pdfplumber/discussions/625#discussioncomment-2368535

Let me know if not, and I can take a closer look at your PDF.

**Reply by Taootrum** [reply]
Hi  jsvine:
Thanks for your reply, I have read [#625(comment)](https://github.com/jsvine/pdfplumber/discussions/625#discussioncomment-2368535) and tried it, but the result is not ideal, as follows:

my code:
```
import pdfplumber

pdf = 'Table_Example.pdf'

with pdfplumber.open(pdf) as pdf:
    page = pdf.pages[0]

    line_pos = max(r["bottom"] for r in page.rects)
    table = page.extract_table({
        "explicit_horizontal_lines": [ line_pos ]
    })

    im = page.to_image()
    line_pos = max(r["bottom"] for r in page.rects)
    im.reset().debug_tablefinder({
        "explicit_horizontal_lines": [ line_pos ]
    }).save('Out.jpg')
```

Demonstrating :
![Out](https://user-images.githubusercontent.com/37130412/160226792-67c12a29-1964-46a2-a320-d753b5547785.jpg)

**Reply by jsvine** [reply]
Ah, yes, I can see why: In the footer of the page, there are rectangles that are *not* part of the table So you'll want a way of identifying the rects that *are* part of your desired table. In the case of *this particular PDF*, one way to do that is by examining/selecting on the rects' `non_stroking_color` attribute:

```python
from collections import Counter
for key, count in Counter(r["non_stroking_color"] for r in page.rects).items():
    print(f"{str(key):<20}: {count}")

#  1                   : 1
#  (0.406, 0.16, 0.523): 1
#  (0.586, 0.41, 0.668): 7
#  0                   : 33
```

It turns out all of the `non_stroking_color == 0` rects are those used to draw the tables:

```python
nsc0_rects = [ r for r in page.rects if r["non_stroking_color"] == 0]
im.reset().draw_rects(nsc0_rects)
```

![image](https://user-images.githubusercontent.com/534702/160645915-549e7f46-0fe1-458f-905b-1dbf5599f109.png)

... and using those in the same approach seems to work:

```python
line_pos = max(r["bottom"] for r in nsc0_rects)

im.reset().debug_tablefinder({
    "explicit_horizontal_lines": [ line_pos ]
})
```

![image](https://user-images.githubusercontent.com/534702/160646167-6420b89b-b0ec-403b-b166-86b682eec1b0.png)

**Reply by Taootrum** [reply]
@jsvine Thanks a lot for your answer, I have successfully parsed the table with your suggestion, thanks again

**Reply by RPASySCreations** [reply]
@Taootrum i have same problem can you please write a code , I am still confuse about ["non_stroking_color"]

**Reply by Taootrum** [reply]
Hi @RPASySCreations ：

Please refer to the following code to successfully parse the attached PDF
```
import pdfplumber

pdf = 'Table_Example.pdf'

with pdfplumber.open(pdf) as pdf:
	page = pdf.pages[0]

	nsc0_rects = [ r for r in page.rects if r["non_stroking_color"] == 0]

	line_pos = max(r["bottom"] for r in nsc0_rects)
	ts = { "explicit_horizontal_lines": [ line_pos ]}

	im = page.to_image()
	im.reset().debug_tablefinder(ts).save('Out.jpg')
```
